src/core/authenticator.cpp

```diff
@@ -8,4 +8,10 @@ Authenticator::Authenticator()
 bool Authenticator::authenticateUser(const std::string& username,
                                      const std::string& password) const {
   return database.loginCheck(username, password);
+}
+
+bool Authenticator::registerUser(const std::string& username,
+                                 const std::string& password,
+                                 const std::string& email) const {
```

Comment:
This one shouldn't be `const` since we are performing an action that significantly impacts code.